report/sections/data.tex

```diff
@@ -1,19 +1,29 @@
+
+% How many samples?
+% How many labels?
+% Distribution of labels in dataset
+There are a total of 453 samples, each with an `.lm2', `.lm3', and `.bnt' file. A single file corresponds to one out of the six possible emotions (anger, disgust, fear, happy, sadness, and surprise). Of these 453 samples, 71 correspond to `anger', 69 to `disgust', 70 to `fear', 106 to `happy', 66 to `sadness', and 71 to `surprise'. Since some labels have more samples than others, the dataset suffers from class imbalance. Class imbalance is a phenomenon which usually results in models which tend to focus on the classification of the samples which are overrepresented while ignoring or misclassifying the underrepresented samples \cite{data-mining-intro}. Section \ref{impl} explains how this issue was addressed.
+
+For each possible file extension a parser was created to extract the features for the emotion classification task. The parsers find all files with a particular extension and create a `.csv' file which consists of all the samples with their corresponding label, and all features associated with each sample.
 
 \subsection{Data Cleansing}
 
-\todo{TODO: Talk about some of the features not being defined in all inputs (NaN)}
+Each sample can have at most 26 features, but it is possible for some of these not to be defined in certain occasions. The parsers address this issue by setting the value of the feature to `NaN' across all the dimensions which the file being parsed defines. In the analysis and classification all features which have at least one undefined (i.e. `NaN') value are removed.
 
 \subsection{Data Transformation}
 
-\todo{TODO: Explain how we tried removing `static' features and computing distance to `Nose Tip' but these ended up not being as accurate as simply using the `regular' features}
+As explained in section \ref{intro:what-has-worked-what-has-not}, a few data transformation strategies were tested. Since none of these achieved positive results, the original features with the removed undefined values were used during the rest of the analysis.
 
 \subsection{Normalization}
 
-\todo{TODO: Why is this necessary?}
+Normalization refers to the process of standardizing the values of independent features of a dataset \cite{data-mining-intro}. Since many of the machine learning techniques use distance to compute the difference between two or more distinct samples, a feature within these samples that has a broad range of values will dominate the process. In order to avoid this, the range of all features was normalized so that each feature contributed approximately proportionately to the computation.
```

Comment:
- [ ] "Since many ~~of the~~ machine learning ..."
- "...values will dominate the process." This is a good sentence but I'm wondering if the subject _machine learning techniques_ gets disconnected from _the process_ which is what you call normalization the sentence before.